src/validation.cpp

```diff
@@ -3040,7 +3047,7 @@ bool ContextualCheckBlock(const CBlock& block, CValidationState& state, const Co
     //   {0xaa, 0x21, 0xa9, 0xed}, and the following 32 bytes are SHA256^2(witness root, witness nonce). In case there are
     //   multiple, the last one is used.
     bool fHaveWitness = false;
-    if (VersionBitsState(pindexPrev, consensusParams, Consensus::DEPLOYMENT_SEGWIT, versionbitscache) == THRESHOLD_ACTIVE) {
+    if (VersionBitsState(pindexPrev, consensusParams, Consensus::DEPLOYMENT_SEGWIT2X, versionbitscache) == THRESHOLD_ACTIVE) {
```

Comment:
What do you mean by "futile"? Segwit is a significant technical upgrade to Bitcoin, not just a blocksize increase.

Because BIP91 is a soft-fork, existing segwit nodes (~80% of the P2P network) are in no particular rush to upgrade, as BIP91 is backwards compatible with segwit.